### PR TITLE
Hollywood modules

### PR DESCRIPTION
**Pull request by jorge07**

### CODE REVIEW

File: src/Infrastructure/Transaction/TransactionModule.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import GetOne from "Application/UseCase/Transaction/GetOne/Handler";
+import TransactionWasCreated from "Domain/Transaction/Events/TransactionWasCreated";
+import Transaction from "Domain/Transaction/Transaction";
+import { EventSourcing, Framework} from "hollywood-js";
+import {SharedModule} from "Infrastructure/Shared/SharedModule";
+import {Transactions} from "Infrastructure/Transaction/ReadModel/Mapping/Transactions";
+import TransactionPostgresProjector from "Infrastructure/Transaction/ReadModel/Projections/TransactionsPostgresProjector";
+import PostgresRepository from "Infrastructure/Transaction/ReadModel/Repository/PostgresRepository";
+import {interfaces} from "inversify";
+import {getRepository} from "typeorm";
```
Comment (joshua-barnett) [inline src/Infrastructure/Transaction/TransactionModule.ts:13]:
```suggestion
import { EventSourcing, Framework} from "hollywood-js";
import { SharedModule } from "Infrastructure/Shared/SharedModule";
import { Transactions } from "Infrastructure/Transaction/ReadModel/Mapping/Transactions";
import TransactionPostgresProjector from "Infrastructure/Transaction/ReadModel/Projections/TransactionsPostgresProjector";
import PostgresRepository from "Infrastructure/Transaction/ReadModel/Repository/PostgresRepository";
import { interfaces } from "inversify";
import { getRepository } from "typeorm";
```

File: src/Kernel.ts [hunk 2]
```diff
@@ -1,16 +1,14 @@
 import { Framework } from "hollywood-js";
+import {TransactionModule} from "Infrastructure/Transaction/TransactionModule";
+import { testParameters } from "../config/paramaters.test";
 import { parameters } from "../config/parameters";
-import { services } from "../config/services";
-import { testServices } from "../config/test";
 
 export default async function KernelFactory(debug: boolean): Promise<Framework.Kernel> {
-     const kernel = await Framework.Kernel.create(
+     return await Framework.Kernel.createFromModuleContext(
```
Comment (joshua-barnett) [inline src/Kernel.ts:7]:
```suggestion
     return Framework.Kernel.createFromModuleContext(
```

File: tests/Application/UseCase/Transaction/Create.test.ts [hunk 3]
```diff
@@ -3,17 +3,17 @@ import TransactionWasCreated from "Domain/Transaction/Events/TransactionWasCreat
 import Transaction from "Domain/Transaction/Transaction";
 import TransactionId from "Domain/Transaction/ValueObject/TransactionId";
 import { Framework } from "hollywood-js";
-import EventCollectorListener from "tests/Infrastructure/Shared/EventCollectorListener";
-import InMemoryTransactionRepository from "tests/Infrastructure/Transaction/InMemoryRepository";
+import {EventCollectorListener} from "tests/Infrastructure/Shared/EventCollectorListener";
```
Comment (joshua-barnett) [inline tests/Application/UseCase/Transaction/Create.test.ts:6]:
```suggestion
import { EventCollectorListener}  from "tests/Infrastructure/Shared/EventCollectorListener";
```

File: config/paramaters.test.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import {Parameter, UniqueParameterIdentifier} from "hollywood-js/src/Framework/Container/Items/Parameter";
```
Comment (joshua-barnett) [inline config/paramaters.test.ts:1]:
```suggestion
import { Parameter, UniqueParameterIdentifier } from "hollywood-js/src/Framework/Container/Items/Parameter";
```

File: tests/TestKernelFactory.ts [hunk 5]
```diff
@@ -76,6 +84,21 @@ export const testServices: Framework.ServiceList = new Map([
...
     ],
 ]);
+
+export async function TestKernelFactory(debug: boolean): Promise<Framework.Kernel> {
+    const TestModule = new Framework.ModuleContext({
+        services: testServices,
+        modules: [TransactionModule]
+    })
+
+    return await Framework.Kernel.createFromModuleContext(
```
Comment (joshua-barnett) [inline tests/TestKernelFactory.ts:97]:
```suggestion
    return Framework.Kernel.createFromModuleContext(
```
Not sure if you need `await` here.

Do you do `return await` everywhere just so it's consistent?

Comment (jorge07) [reply tests/TestKernelFactory.ts:97]: Yep but it's redundant, doesn't make sense really

**Pull request merged**